ff_dir.c

```diff
@@ -2875,24 +2875,31 @@ FF_Error_t FF_ExtendDirectory( FF_IOManager_t * pxIOManager,
 } /* FF_ExtendDirectory() */
 /*-----------------------------------------------------------*/
 
-static const uint8_t forbiddenChrs[] =
+/* *INDENT-OFF* */
+#if ( ffconfigUNICODE_UTF16_SUPPORT != 0 )
+    static const FF_T_WCHAR forbiddenChrs[] =
+#else
+    static const uint8_t forbiddenChrs[] =
+#endif
 {
 /* Windows says: don't use these characters: '\/:*?"<>|'
  *  "     *     /	    :     <     >     ?    '\'    ?     | */
     0x22, 0x2A, 0x2F, 0x3A, 0x3C, 0x3E, 0x3F, 0x5C, 0x7F, 0x7C
 };
+/* *INDENT-ON* */
 
 /* *INDENT-OFF* */
 #if ( ffconfigUNICODE_UTF16_SUPPORT != 0 )
-    static void FF_MakeNameCompliant( FF_T_WCHAR * pcName )
+    static FF_Error_t FF_MakeNameCompliant( FF_T_WCHAR * pcName )
 #else
-    static void FF_MakeNameCompliant( char * pcName )
+    static FF_Error_t FF_MakeNameCompliant( char * pcName )
 #endif
 /* *INDENT-ON* */
 {
+    FF_Error_t xReturn = pdTRUE;
```

Comment:
Also suggest not having an xReturn just use the fixed value in your `return` statement.  Only use xReturn if you need to decide on the return value and still perform some actions.  - i.e. prefer early returns in your code but I don't know the full `coding guidelines` for this repo so this style might not be preferred.

```suggestion
FF_Error_t xReturn = FF_ERR_NONE;
```

Reply:
I would prefer an early return in this function too.
Is it allowed?

Reply:
Early returns are not allowed by the MISRA rules. Sometimes you see this construction:
~~~c
void foo()
{
    FF_Error_t xReturn = FF_ERR_NONE;

    /* This while loop is only introduced to be able to use break. */
    do
    {
        xError = FF_CreateDirent();
        if( FF_isERR( xReturn ) == pdFALSE )
        {
            break;
        }
    }
    while( pdFALSE );

    return xReturn ;
}
~~~
So in a way, this also looks like an early return.

The advantage: when debugging, you can put a break on the one and only return statement.